pkg/apis/kops/cluster.go

```diff
@@ -110,6 +110,8 @@ type ClusterSpec struct {
 	UpdatePolicy *string `json:"updatePolicy,omitempty"`
 	// Additional policies to add for roles
 	AdditionalPolicies *map[string]string `json:"additionalPolicies,omitempty"`
+	// EnableEtcdTLS indicates the etcd service should use TLS between peers and clients
+	EnableEtcdTLS bool `json:"enableEtcdTLS,omitempty"`
```

Comment:
I agree that it's awkward having to set it twice... I would have guessed that setting the override to http (instead of https) would have worked, but I can easily see that this isn't a scenario that k8s would have tested.

The whole split of events is sort of vestigial anyway.

My 2c is that because it is more logically coherent we should put the TLS config on the EtcdClusterSpec, and raise errors in validation if the value is non-equal across clusters.  I imagine we would do the same for etcd versions.  I don't think there's a big use case for setting them separately, but I also don't think it's something that will be changed terribly often so I think it's a tolerable inconvenience.

I think in general we want to get people to TLS & etcd3.  One way to do that have create cluster default new  clusters to TLS with etcd3, as of a certain version of k8s.

When we have aggregated api servers, we might have more etcd servers.  They likely won't be configured in the same Cluster object, but it would be nice here to have a consistent self-contained object.

But this is my opinion - if you disagree we can do it the other way!